Change delete button for routes to delete all routes for page

thomasiles opened the pull request:
## Let form creators delete all routes for a question

https://trello.com/c/KA2MdRKW/2036-change-delete-routes-button-to-delete-all-routes-for-a-page

https://app.mural.co/t/gaap0347/m/gaap0347/1728478914347/6a7e0b709f50f00f81cab37414fdf44ec8601a70

The delete button on the routes#show page now takes the user to a confirmation screen for removing all the routes associated with the page, not just the primary route.

# screenshots
![image](https://github.com/user-attachments/assets/c202aba2-82c7-4257-b3e0-8e99c301b0df)

![image](https://github.com/user-attachments/assets/9f29ba45-a100-46eb-88e1-b495412478db)

![image](https://github.com/user-attachments/assets/caaeaa10-9f64-4a22-845e-0e64387c48f6)

thomasiles commented on app/controllers/pages/routes_controller.rb:
```diff
@@ -3,4 +3,40 @@ def show
+      rc.prefix_options[:form_id] = current_form.id
+      rc.prefix_options[:page_id] = rc.routing_page_id
+      ConditionRepository.destroy(rc)
+    end
```
This doesn't feel like the best place for this. I'm not sure where is though?

We need to get the form, then every page and then delete the conditions.

It could go in one of the repositories but I'm not 100% sure which one is the best choice. 

It could be a separate service or a new RouteRepository itself?

lfdebrux replied:
The idea of a route repository is interesting!

I wonder about putting it in the form input object though, that might make sense? I guess we would need a new one, but it could inherit from `DeleteConfirmationInput` to avoid too much code duplication?

lfdebrux commented on app/controllers/pages/routes_controller.rb:
```diff
@@ -3,4 +3,40 @@ def show
+    @delete_confirmation_input = Pages::Routes::DeleteConfirmationInput.new(delete_confirmation_input_params)
+
+    if @delete_confirmation_input.invalid?
+      return render :delete, locals: { current_form:, page:, delete_confirmation_input: @delete_confirmation_input }
```
This should probably have `status: :unprocessable_entity`?

hannahkc commented on config/locales/en.yml:
```diff
@@ -1145,6 +1150,11 @@ en:
+    routes:
+      delete:
+        back: Back to question %{page_number}’s routes
+        caption: Question %{page_number}'s routes
```
```suggestion
        caption: Question %{page_number}’s routes
```

thomasiles commented:
I've moved the delete routes function and added a better test for it.

I've also corrected all the dodgy punctuation. I swear I check them every time and every time I get them wrong.

DavidBiddle reviewed (approved):
The code changes look good to me and I can see the deletion happening as expected when I run this locally 🎉
